mmdet3d/core/bbox/coders/point_xyzwhlr_bbox_coder.py

```diff
@@ -0,0 +1,110 @@
+import numpy as np
+import torch
+
+from mmdet.core.bbox import BaseBBoxCoder
+from mmdet.core.bbox.builder import BBOX_CODERS
+
+
+@BBOX_CODERS.register_module()
+class PointXYZWHLRBBoxCoder(BaseBBoxCoder):
+    """Point based bbox coder for 3D boxes.
+
+    Args:
+        code_size (int): The dimension of boxes to be encoded.
+        use_mean_size (bool, optional): Whether using anchors based on class.
+            Defaults to True.
+    """
+
+    def __init__(self, code_size=7, use_mean_size=True, **kwargs):
+        super(PointXYZWHLRBBoxCoder, self).__init__()
+        self.code_size = code_size
+        self.use_mean_size = use_mean_size
+        if self.use_mean_size:
+            self.mean_size = torch.from_numpy(np.array(
+                kwargs['mean_size'])).cuda().float()
```

Comment:
.cuda() is dangerous. Can we get the device information from input arguments?

Reply:
it is in initial step, so we are not get to get the device information from input arguments.

Reply:
should use to(xxx.device) in encode/decode process